src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/VariableDeclarationUsageDistanceCheck.java

```diff
@@ -810,11 +810,15 @@ private static DetailAST getFirstNodeInsideIfBlock(
     private static DetailAST getFirstNodeInsideSwitchBlock(
             DetailAST block, DetailAST variable) {
         final DetailAST currentNode = getFirstCaseGroupOrSwitchRule(block);
+        int currentNodeType = -1;
```

Comment:
Pretty weird code.
Do you try to skip `TokenUtil.forEachChild(block, currentNodeType` ?

If yes, please use IF.

Reply:
I think I figured it out, working on the tests right now. Just trying different stuff.

Reply:
I will tag you when everything is done.

Reply:
@romani Should be okay now.